### PR TITLE
Implement requirement for manual install of CBMs

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
SUMMARY: Infrastructure "Implement installation requirment for CBMs"

#### Purpose of change
You can now specify a requirement_id to set the tools and components necessary to manualy install a CBM

#### Describe the solution

Add requirement_id entry to bionic_data
Check for requirement when activating a CBM
     No requirement means manual install is not possible
If the player has the tool and components consume them and porceed with the installation

#### Testing

Give sewing_standard to finger laser cbm
Get a sewing tool and thread
Try to install
Get popup about missing cutting quality
Get a knife an retry
Popup about missing anesthetic
Get anesthesia kit and retry
Porceed with instalation and get a faulty install
Anesthetic was consumed
A thread was consumed

Test that tacoma doctor still works with or without requirement

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
We've done such a large work to get rid of manual installation of CBMs, and now we're returning back?

**Comment by Fris0uman** [comment]
This system can be defined per CBM instead of the all or nothing we had before, also I don't really plan on making any current vanilla CBM installable manually, but maybe aftershock could use this and it could also be the basis for having NPCs performing surgery on the player.

**Comment by Rail-Runner** [comment]
Could we get the same thing for CBM removal? I think it could be particularly useful for mods adding CBMs that are not actually bionics and rather some other kind of permanent non-random body modification.

**Comment by Fris0uman** [comment]
There's no support in place for manual uninstallation and I'm not interested in adding it.

File: src/bionics.cpp [hunk 1]
```diff
@@ -280,6 +280,13 @@ void bionic_data::load( const JsonObject &jsobj, const std::string )
 
     optional( jsobj, was_loaded, "available_upgrades", available_upgrades );
 
+    if( jsobj.has_string( "installation_requirement" ) ) {
+        std::string temp_string;
+        optional( jsobj, was_loaded, "installation_requirement", temp_string );
+        instalation_requirement = requirement_id( temp_string );
+    }
+
+
```
Comment (KorGgenT) [inline src/bionics.cpp:289]:
```suggestion
    optional( jsobj, was_loaded, "installation_requirement", instalation_requirement );
```
that misspelling of installation_requirement in the variable name is going to bug me. also, this should work for string_ids just fine.

File: src/bionics.h [hunk 2]
```diff
@@ -126,6 +126,9 @@ struct bionic_data {
      */
     std::set<bionic_id> available_upgrades;
 
+    /**Requirement to bionic installation*/
+    requirement_id instalation_requirement;
```
Comment (KorGgenT) [inline src/bionics.h:130]: installation is misspelled here

**Pull request merged**